pagefaultgames/pokerogue
# [Enhancement] Fixes sizing and legacy theme problem for bgm-bar

## What are the changes?
Fixes sizing and legacy theme problem for bgm-bar

## Why am I doing these changes?
Depending on the locale chosen, the text could be overlaid, not to mention over-indented if the translation of "Music" is longer.
We also discovered with @brain-frog that the bgm-bar has not been tested at all with the legacy theme, and that on the latter, it is simply unusable as such.

## What did change?
- Adds TextStyle for bgm-bar ui (To define it properly with the others)
- Makes the ui bgm-bar truly dynamic (no overlap) + cleanup unused code
  > Removes all hard-coded characters from the string to allow for different typographical rules in different languages (Some write `word : ` and others `word: `, for example)
- Applies the necessary modifications to locales following the deletion of the hard-coded character to form strings
- Creates and loads a resource for the bgm-bar ui
  > Can be redesigned by someone with more talent if needed, since it won't affect the ability bar

### Screenshots/Videos
> Before
![image](https://github.com/pagefaultgames/pokerogue/assets/8146474/bdc3574c-0511-406d-ba41-cdf32a1b05a6)

> After
![image](https://github.com/pagefaultgames/pokerogue/assets/8146474/07715c24-e900-4739-a010-4b5f2e6d2124)

## How to test the changes?
Set the BGM display in the options, and watch the result when the menu is opened.
(_Note that I've noticed a bug when the option is activated: sometimes you have to press F5 for the text to be displayed correctly, and a giant box with no text may appear. However, as the bug is already present, and is not the subject of this PR, I'll look into it later._)
![bug_prod](https://github.com/pagefaultgames/pokerogue/assets/8146474/0568b644-f171-42a6-83a9-ab354f15fc73)

## Checklist
